whipper/image/table.py

```diff
@@ -443,30 +408,37 @@ def _getMusicBrainzValues(self):
         # number of first track
         result.append(1)
 
-        # number of last audio track
-        result.append(self.getAudioTracks())
+        # number of last audio track (default: number of audio tracks)
+        lastTrack = self.getAudioTracks()
+        result.append(lastTrack)
 
-        leadout = self.leadout
-        # if the disc is multi-session, last track is the data track,
-        # and we should subtract 11250 + 150 from the last track's offset
-        # for the leadout
-        if self.hasDataTracks():
-            assert not self.tracks[-1].audio
-            leadout = self.tracks[-1].getIndex(1).absolute - 11250 - 150
-
-        # treat leadout offset as track 0 offset
-        result.append(150 + leadout)
+        dataTrackLast = False
+        additional = 0
+        offsets = []
 
         # offsets of tracks
-        for i in range(1, 100):
-            try:
-                track = self.tracks[i - 1]
-                if not track.audio:
-                    continue
-                offset = track.getIndex(1).absolute + 150
-                result.append(offset)
-            except IndexError:
-                pass
+        for i in range(0, len(self.tracks)):
+            track = self.tracks[i]
+            if not track.audio:
+                # if the data track is not at the end
+                if i < len(self.tracks) - 1:
+                    additional += 1
+                else:
+                    # if the data track is last
+                    dataTrackLast = True
+                    sectors = self.tracks[-1].getIndex(1).absolute - 11400
+                    # treat leadout offset as track 0 offset
+                    sectors += 150
+                continue
+            offset = track.getIndex(1).absolute + 150
+            offsets.append(offset)
+
+        if not dataTrackLast:
+            # the end of the last audio track
+            sectors = self.getTrackEnd(lastTrack + additional) + 1 + 150
```

Comment:
Why `+ 1 + 150` here?

Reply:
+1 because getTrackEnd returned value is always 1 unit down and +150 is just similar to how we calculate offsets. Which means thats actually `offsets[-1] + getTrackLength(lastTrack)`

 I think i can come up with a more general formula to make it less confusing.

Reply:
I think just adding a short comment to explain this would be fine.

Reply:
@JoeLametta , what do you think about the new comment? (i think it's a little short tho). ;)